File: 11/js/part1.js
const fs = require('fs')

const contents = fs.readFileSync('../input.txt', 'utf8')
const lengths = contents.split('\n').filter(line => line !== '')[0].split(',')

let xOffset = 0
let yOffset = 0

for (let length of lengths) {
  switch (length) {
    case 'n':
      yOffset += 1
      break
    case 's':
      yOffset -= 1
      break
    case 'ne':
      yOffset += 0.5
      xOffset += 0.5
      break
    case 'nw':
      yOffset += 0.5
      xOffset -= 0.5
      break
    case 'se':
      yOffset -= 0.5
      xOffset += 0.5
      break
    case 'sw':
      yOffset -= 0.5
      xOffset -= 0.5
      break
    default:
  }
}

console.log(Math.abs(xOffset) + Math.abs(yOffset))

File: 13/ex/part1.ex
defmodule Part1 do
  def solve do
    {:ok, contents} = File.read "../input.txt"
    contents
    |> String.trim
    |> String.split("\n")
    |> Enum.map(&parse_line/1)
    |> compute_severity
    |> IO.inspect
  end

  defp compute_severity (layers) do
    layers
    |> Enum.filter(&caught?/1)
    |> Enum.map(&calc_severity/1)
    |> Enum.reduce(&+/2)
  end

  defp caught? ([layer_num, depth]) do
    rem(layer_num, 2 * (depth - 1)) == 0
  end

  defp calc_severity ([layer_num, depth]) do
    layer_num * depth
  end

  defp parse_line (line) do
    String.split(line, ": ")
    |> Enum.map(&String.to_integer/1)
  end
end
